### PR TITLE
Cleanup assertions, convert to macros where possible.

### PR DESCRIPTION
**Pull request by nlutsenko**
Improved many places where we do weird assertions, or where we use a condition but could have used a macro. Also fixed some typos.
Improves the bigger enhancement request in #6

### CODE REVIEW

File: Parse/PFRole.m [hunk 1]
```diff
@@ -62,28 +64,17 @@ - (PFRelation *)users {
...
-        if (![object isKindOfClass:[NSString class]]) {
-            [NSException raise:NSInvalidArgumentException
-                        format:@"A role's name must be an NSString."];
-        }
-        if ([object rangeOfString:@"^[0-9a-zA-Z_\\- ]+$" options:NSRegularExpressionSearch].location == NSNotFound) {
-            [NSException raise:NSInvalidArgumentException
-                        format:@"A role's name can only contain alphanumeric characters, _, -, and spaces."];
-        }
+    if ([key isEqualToString:@keypath(PFRole, name)]) {
+        PFConsistencyAssert(self.objectId, @"A role's name can only be set before it has been saved.");
```
Comment (richardjrossiii) [inline Parse/PFRole.m]: Wrong.

File: Parse/PFRole.m [hunk 2]
```diff
@@ -62,28 +64,17 @@ - (PFRelation *)users {
...
     }
     [super setObject:object forKey:key];
 }
 
 - (BFTask *)saveInBackground {
-    if (!self.objectId && !self.name) {
-        [NSException raise:NSInternalInconsistencyException
-                    format:@"New roles must specify a name."];
-    }
+    PFConsistencyAssert(!self.objectId || self.name, @"New roles must specify a name.");
```
Comment (richardjrossiii) [inline Parse/PFRole.m]: This seems wrong as well. Shouldn't it be

    self.objectId || self.name

?

**Comment by richardjrossiii** [comment]
Too much boolean logic for a friday evening :)

**Comment by nlutsenko** [comment]
Too much indeed...

**Pull request merged**